lib/OpenQA/WebAPI/Controller/API/V1/Webhook.pm

```diff
@@ -98,16 +98,14 @@ sub product ($self) {
       unless $SUPPORTED_PR_ACTIONS{$action};
     my $pr = $json->{pull_request} // {};
     my $head = $pr->{head} // {};
-    my $sha = $head->{sha};
     my $repo = $head->{repo} // {};
-    my $repo_name = $repo->{full_name};
-    my $clone_url = $repo->{clone_url};
-    my $statuses_url = $pr->{statuses_url};
-    my $html_url = $pr->{html_url};
-    return $self->render(
-        status => 400,
-        text => '"pull_request" lacks "statuses_url", "head/sha" or "head/repo/full_name" or "head/repo/clone_url"'
-    ) unless $sha && $repo_name && $clone_url && $statuses_url;
+    my @missing;
+    push @missing, 'pull_request/id' unless my $pr_id = $pr->{id};
```

Comment:
I'd favor less procedural code but I guess you would have applied a "map" approach or something different if you would have known a good way :)

Reply:
I tried but ended up with that code as the best solution. Here is what I was thinking/considering: This approach has the advantage of being not complicated at all, one has the required fields directly available as local variables and the error message that is computed is also very specific. There is also just one line for each variable which is acceptable. Better than one line for each variable (e.g. a map definition that's too long for one line) *plus* some extra helper code to make it actually do anything. I guess the next best approach would be to define a schema and use the JSON validator but I personally find it more complicated to understand as one then need to understand the abstraction to understand what the code does. It would probably be also more lines of code and definitely less efficient.